server/chat-commands/core.ts

```diff
@@ -1237,6 +1237,9 @@ export const commands: ChatCommands = {
 		if (Punishments.isBattleBanned(user)) {
 			return this.popupReply(`You are banned from battling and cannot challenge users.`);
 		}
+		if (!user.named || user.namelocked) {
+			return this.popupReply(`Unnamed users cannot challenge.`);
+		}
```

Comment:
```suggestion
		if (!user.named) {
			return this.popupReply(`You must choose a username before you challenge someone.`);
		}
```

Namelocked users should be able to challenge other locked/namelocked users but no one else; this was discussed elsewhere.

Reply:
Should i add that here, as well?

Reply:
New PR also works.

Reply:
Okay, then!
I had that committed so i guess i'll just push and PR.